ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -251,6 +253,12 @@ public void start() throws StoreException {
         index = new PersistentIndex(dataDir, storeId, taskScheduler, log, config, factory, recovery, hardDelete,
             diskIOScheduler, metrics, time, sessionId, storeDescriptor.getIncarnationId());
         compactor.initialize(index);
+        if (config.storeRebuildTokenBasedOnCompactionHistory
+            && config.storePartitionsToRebuildTokenBasedOnCompactionHistory.contains(
+            replicaId.getPartitionId().getId())) {
+          compactor.enablePersistIndexSegmentOffsets();
```

Comment:
this config is used to enable rebuild token based con compaction history, but here we also use it to enable new version of compaction log, which means, the version of compaction log would only be created for those partitions that enables this feature.